### PR TITLE
solve: allow Implication to have multiple conditions

### PR DESCRIPTION
**Pull request by zmedico**
@mgorny

### CODE REVIEW

File: parser.py [hunk 1]
```diff
@@ -31,10 +31,11 @@ def negated(self):
 
 
 class Implication(object):
-    def __init__(self, condition, constraint):
+    def __init__(self, condition, constraint, multi_condition=False):
```
Comment (mgorny) [inline parser.py:34]: Given that such implications can not be produced by any code in the suite, are not covered by the specification and breaks assumptions made in other parts of the code, that's a big :-1:. This is the reference implementation that means to visualize and test the specification, and it's not going beyond that.

**Comment by mgorny** [comment]
Cherry-picked the second commit.

File: parser.py (hunk 1 above)
Comment (zmedico) [reply parser.py:34]: This code from flatten3 is what triggered it for me:
```python
		elif isinstance(expr, AnyOfOperator):
			# || ( a b c ... ) -> [!b !c ...]? ( a )
			yield (conditions+[x.negated() for x in expr.constraint[1:]], expr.constraint[0])
```
If anyone wants to test solve.py with `|| ( a b c )` then they'll hit the assertion.

Comment (mgorny) [reply parser.py:34]: Ok, you're right. I've removed the assert entirely then. Thanks.

Comment (zmedico) [reply parser.py:34]: Cool!

**Pull request closed without merging**